From 7b1b018826810b99f9f697904001b1c13d9e1b04 Mon Sep 17 00:00:00 2001
From: Peter Simons <simons@cryp.to>
Date: Thu, 29 Feb 2024 15:45:22 +0100
Subject: [PATCH] Language.Docker: enable DuplicateRecordFields to fix build
 error with ghc-9.8.x
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

src/Language/Docker.hs:26:5: error: [GHC-97219]
    Duplicate record field ‘sourcePaths’ in export list:
       ‘Language.Docker.Syntax.AddArgs(..)’ exports the field ‘Language.Docker.Syntax.sourcePaths’
       belonging to the constructor ‘Language.Docker.Syntax.AddArgs’
         imported qualified from ‘Language.Docker.Syntax’ at src/Language/Docker.hs:46:1-39
         (and originally defined at src/Language/Docker/Syntax.hs:193:9-19)
       ‘Language.Docker.Syntax.CopyArgs(..)’ exports the field ‘Language.Docker.Syntax.sourcePaths’
       belonging to the constructor ‘Language.Docker.Syntax.CopyArgs’
         imported qualified from ‘Language.Docker.Syntax’ at src/Language/Docker.hs:46:1-39
         (and originally defined at src/Language/Docker/Syntax.hs:174:9-19)

Fixes https://github.com/hadolint/language-docker/issues/93.
---
 src/Language/Docker.hs | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/Language/Docker.hs b/src/Language/Docker.hs
index 382118d..251ba38 100644
--- a/src/Language/Docker.hs
+++ b/src/Language/Docker.hs
@@ -1,3 +1,5 @@
+{-# LANGUAGE DuplicateRecordFields #-}
+
 module Language.Docker
   ( Language.Docker.Syntax.Dockerfile,
 
